server/models/procedures/trial_balance.sql

```diff
@@ -319,22 +319,23 @@ BEGIN
   DELETE FROM posting_journal WHERE record_uuid IN (SELECT record_uuid FROM stage_trial_balance_transaction);
 
   -- Let specify that this invoice or the cash payment is posted
-  SELECT COUNT(uuid) INTO isInvoice  FROM invoice  WHERE invoice.uuid IN (SELECT record_uuid FROM stage_trial_balance_transaction);
-  SELECT COUNT(uuid) INTO isCash  FROM cash  WHERE cash.uuid IN (SELECT record_uuid FROM stage_trial_balance_transaction);
-  SELECT COUNT(uuid) INTO isVoucher  FROM voucher  WHERE voucher.uuid IN (SELECT record_uuid FROM stage_trial_balance_transaction);
+  SELECT COUNT(uuid) INTO isInvoice FROM invoice WHERE invoice.uuid IN (SELECT record_uuid FROM stage_trial_balance_transaction);
+  SELECT COUNT(uuid) INTO isCash FROM cash WHERE cash.uuid IN (SELECT record_uuid FROM stage_trial_balance_transaction);
+  SELECT COUNT(uuid) INTO isVoucher FROM voucher WHERE voucher.uuid IN (SELECT record_uuid FROM stage_trial_balance_transaction);
 
   IF isInvoice > 0 THEN
-    UPDATE invoice SET posted = 1 WHERE uuid IN (SELECT record_uuid FROM stage_trial_balance_transaction);
+    UPDATE invoice SET posted = 1 WHERE EXISTS (SELECT 1 FROM stage_trial_balance_transaction t WHERE t.record_uuid = invoice.uuid);
```

Comment:
SQL reminds me of COBOL.  There may be some folks that are completely fluent in it; but the huge majority of folks who dabble in it have to look up every syntactical thing and then still hack around!  (You might infer that I know COBOL from this; you would be wrong!)

I have no idea why you need to do it this way!  If this necessary in general going forward, I suggest sending the team an email with details.